impl_linux.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+// +build linux
+
+package mutex
+
+import (
+	"net"
+	"path/filepath"
+
+	"github.com/juju/errors"
+)
+
+const prefix = "@/var/lib/juju/mutex-"
+
+type impl struct {
+	socket *net.UnixListener
+}
+
+func acquire(name string) (Releaser, error) {
+	path := filepath.Join(prefix, name)
+	addr, err := net.ResolveUnixAddr("unix", path)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	l, err := net.ListenUnix("unix", addr)
+	if err != nil {
+		return nil, errors.Wrap(err, ErrLocked)
+	}
+	return &impl{socket: l}, nil
+}
+
+// Release implements Releaser.
+func (i *impl) Release() error {
+	return i.socket.Close()
```

Comment:
I agree we shouldn't double-close the socket, but why do we need to panic instead of just informing the client that  they're an idiot?